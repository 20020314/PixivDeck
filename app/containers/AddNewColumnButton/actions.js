// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
