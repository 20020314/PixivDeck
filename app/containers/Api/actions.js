// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
