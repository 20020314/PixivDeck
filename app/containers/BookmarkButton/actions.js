// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
