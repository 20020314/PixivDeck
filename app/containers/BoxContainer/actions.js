// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
