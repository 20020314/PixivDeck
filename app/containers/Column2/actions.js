// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
