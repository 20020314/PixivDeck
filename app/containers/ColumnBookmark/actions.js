// @flow
import type { Action } from './actionTypes.js'
import {
  ADD_BOOKMARK_COLUMN,
  ADD_BOOKMARK_COLUMN_SUCCESS,
  FETCH_BOOKMARK,
  FETCH_BOOKMARK_FAILRE,
  FETCH_NEXT_BOOKMARK,
  FETCH_NEXT_BOOKMARK_FAILRE,
  SET_NEXT_URL,
  FETCH_BOOKMARK_SUCCESS,
  FETCH_NEXT_BOOKMARK_SUCCESS,
} from './constants'
import type { Response } from 'services/api'
import type { ColumnId } from './reducer'

export function addBookmarkColumn(id: ColumnId): Action {
  return {
    type: ADD_BOOKMARK_COLUMN,
    id,
  }
}

export function addBookmarkColumnSuccess(id: ColumnId): Action {
  return {
    type: ADD_BOOKMARK_COLUMN_SUCCESS,
    id,
  }
}

export function fetchBookmark(id: ColumnId): Action {
  return {
    type: FETCH_BOOKMARK,
    id,
  }
}

export function fetchBookmarkFailre(id: ColumnId): Action {
  return {
    type: FETCH_BOOKMARK_FAILRE,
    id,
  }
}

export function fetchNextBookmark(id: ColumnId): Action {
  return {
    type: FETCH_NEXT_BOOKMARK,
    id,
  }
}

export function fetchNextBookmarkFailre(id: ColumnId): Action {
  return {
    type: FETCH_NEXT_BOOKMARK_FAILRE,
    id,
  }
}

export function setNextUrl(id: ColumnId, nextUrl: string): Action {
  return {
    type: SET_NEXT_URL,
    id,
    nextUrl,
  }
}

export function fetchBookmarkSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<string>
): Action {
  return {
    type: FETCH_BOOKMARK_SUCCESS,
    id,
    response,
    ids,
  }
}

export function fetchNextBookmarkSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<string>
): Action {
  return {
    type: FETCH_NEXT_BOOKMARK_SUCCESS,
    id,
    response,
    ids,
  }
}
