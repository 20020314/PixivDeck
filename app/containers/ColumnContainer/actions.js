// @flow
import type { Action } from './actionTypes.js'
import { CLOSE } from './constants'

export function close(id: number): Action {
  return {
    type: CLOSE,
    id,
  }
}
