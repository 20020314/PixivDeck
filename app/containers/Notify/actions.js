// @flow
import { ADD_NOTIFY, ADD_NOTIFY_W_ITH_ILLUST } from './actionTypes'
import type { AddNotify, AddNotifyWIthIllust } from './actionTypes'

export function addNotify(): AddNotify {
  return {
    type: ADD_NOTIFY,
  }
}
export function addNotifyWIthIllust(
  title: string,
  id: number
): AddNotifyWIthIllust {
  return {
    type: ADD_NOTIFY_W_ITH_ILLUST,
    title,
    id,
  }
}
