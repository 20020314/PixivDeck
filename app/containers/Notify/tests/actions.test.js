// @flow
import * as actions from '../actions'

test('addNotify', () => {
  expect(actions.addNotify()).toMatchSnapshot()
})

test('addNotifyWithIllust', () => {
  expect(actions.addNotifyWithIllust('new illust', 1)).toMatchSnapshot()
})
