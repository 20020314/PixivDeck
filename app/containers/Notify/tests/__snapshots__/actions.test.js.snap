// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addNotify 1`] = `
Object {
  "type": "Notify/add",
}
`;

exports[`addNotifyWithIllust 1`] = `
Object {
  "id": 1,
  "title": "new illust",
  "type": "Notify/ADD_NOTIFY_WITH_ILLUST",
}
`;
